control-plane/agents/src/bin/core/pool/service.rs

```diff
@@ -182,7 +184,13 @@ impl Service {
                 }
                 pools
             }
-            None => self.registry.get_node_opt_pools(node_id).await?,
+            None => match node_id {
```

Comment:
What if pool_id is Some?

Reply:
created a function just to filter if volume_id is given.  It has no connection with node-id or pool-id